### ISSUE TITLE
Trying to access to a non-existing or a non-owned site hangs indefinitely  

### ISSUE THREAD

**Issue by Akronix** [wp-pwa/worona-dashboard#67]
If the user tries to access to a screen with a non-existing site id, let's say:
http://dashboard.worona.org/check-site/foo , then the interface hangs saying "Retrieving data..." indefinitely.
Same happens if user tries to access to a deleted site or to a site which she doesn't own.
A possible solution would be to add a timeout and redirect to /sites in case that the page has spent too much time to render.
Another possible solution would be to check against the server if the site id exists and is owned by current user.

**Comment by Poliuk** [comment]
I think that the second approach is better.

> check against the server if the site id exists and is owned by current user .

And display an error message in the client

**Comment by Akronix** [comment]
The drawback of doing that is we need to do an extra call to the server just to verify if the URL is valid for this user.

**Comment by Poliuk** [comment]
Can't we do it in the same server call used to retrieve the data? 

The server instead of not answering could answer with an error message. 

I'm not sure if this works with our current architecture.

**Comment by Akronix** [comment]
Well, It isn't exactly like server is not answering to a request or anything like that. This what happens:
- subscriptions package: Server retrieves the subscriptions data owned by logged in user, when user logs in.
- bulma package: UI tries to show check-site screen for `foo` and blocks ("retrieving data") until foo is loaded in the client.

As user doesn't own `foo` (actually foo isn't even an existing siteId ), the site `foo` never arrives to the client and UI waits indefinitely.
These two are different packages not related to each other, they both run in parallel and are not directly connected one to the other.

The second approach involves to do a specific server call in order to confirm if the URL is for a siteId that exists and user owns. For this to be consistent, it'd have to be made in every screen that the app has an URL pointing to a siteId. But note that if the URL is OK, this call isn't necessary at all since eventually the client will get the site subscription and the UI will be automatically refreshed.

Perhaps the best solution would be a combination of both: We could add a timeout and when the time out expires, if no requested site has been retrieved, it can trigger a check against the server in order to know if it is a valid URL for the current user.

Moreover, there's a another possible scenario where user has dropped in a very bad connection. In this case, the requested URL is OK for the user but the site subscription never arrives. We could take advantage of the solution above to check also if what happens in reality is that user has dropped to a very bad connection, even when the URL is right.

Anyways we should leave this until First Lauch :rocket: ; because this isn't a critical bug and the solution can change according the app grows and the architecture evolves.

**Comment by luisherranz** [comment]
Won't fix ATM.
